Formula/buildkit.rb

```diff
@@ -0,0 +1,21 @@
+class Buildkit < Formula
+  desc "Сoncurrent, cache-efficient, and Dockerfile-agnostic builder toolkit"
+  homepage "https://github.com/moby/buildkit"
+  url "https://github.com/moby/buildkit/archive/v0.3.3.tar.gz"
+  sha256 "07a44de2bcbaa24055d1a7c9223be9483eb5ec68bf79c9e51c7d583a9b430820"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/moby/buildkit/").install Dir["*"]
+
+    ldflags = ["-X github.com/moby/buildkit/version.Version=#{version}",
+               "-X github.com/moby/buildkit/version.Package=github.com/moby/buildkit"]
+    system "go", "build", "-o", bin/"buildctl", "-ldflags", ldflags.join(" "), "github.com/moby/buildkit/cmd/buildctl"
+  end
+
+  test do
+    system "#{bin}/buildctl", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
**buildkit** itself consist of 2 parts: a daemon **buildd** and a client **buildctl** (very similar to docker model). The daemon doesn't support MacOS, but client does (see https://github.com/moby/buildkit/issues/132). So this formula consist of client only and it's hard to find a way how to test it properly without using docker, for example.

Reply:
Can you make it try some operation, and check that it fails?

Reply:
That doesn't sound like a reasonable thing to test.
Checking that the version command returns non-zero looks good to me.

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
This is the safest thing I can think of:
```bash
buildctl --addr unix://dev/null --timeout 0 build
  # check return code != 0
```

Reply:
Thanks for the suggestion, I'll add it